hack/validate-canary.sh

```diff
@@ -152,17 +155,36 @@ function run() {
     clean-canary
     generate-canary-manifest
     deploy-canary
-    # Tail the logs of the reporter to block until it completes. The report only logs the result file.
-    res=$(kubectl logs -f oci-cloud-controller-manager-canary -c oci-cloud-controller-manager-canary-test-reporter)
-    # Display the results.
-    echo "${res}"
-    # Grep the log to return an error code.
-    error=$(echo "${res}" | grep 'end_time' | cut -d':' -f 1)
-    if [ -z ${error} ]; then
-        exit 1
-    else
-        exit 0
-    fi
+
+    local canary_runs=${CANARY_RUNS}
+    local duration=1800
+    local sleep=10
+    local timeout=$(($(date +%s) + $duration))
+    while [ $(date +%s) -lt $timeout ]; do
+        echo "waiting for ${canary_runs} runs."
+        local logs=$(kubectl logs oci-cloud-controller-manager-canary -c oci-cloud-controller-manager-canary-test-reporter)
+        local num_runs=$(echo "${logs}"| grep 'end_time' | uniq | wc -l)
+        echo "currently run ${num_runs} times."
+        if [ "${num_runs}" -ge "${canary_runs}" ]; then
+            # Remove canary and delete any remaining test namespaces.
+            kubectl delete pod oci-cloud-controller-manager-canary
+            local res=$(kubectl get ns | grep 'cm-e2e-tests-' | awk '{print $1}')
+            if [ ! -z "${res}" ]; then
+                echo ${res} | xargs kubectl delete ns
+            fi
+            #  Test results
+            local num_pass=$(echo "${logs}"| grep '"create_lb": "1"' | uniq | wc -l)
+            local num_fail=$(echo "${logs}"| grep '"create_lb": "0"' | uniq | wc -l)
+            if [ "${num_fail}" -gt "0" ]; then 
+                echo "FAILED"
```

Comment:
Maybe dump the test logs here for the failure case, i.e. it might make debugging a failure easier?